packages/crypto/tests/wycheproof_secp256k1.rs

```diff
@@ -0,0 +1,412 @@
+                }
+                _ => panic!("Found unexpected result value"),
+            }
+            if tc.result == "valid" {}
+        }
+    }
+    assert_eq!(tested, number_of_tests);
+}
+
+#[test]
+fn ecdsa_secp256k1_sha3_512() {
+    let mut tested: usize = 0;
+    let File {
+        number_of_tests,
+        test_groups,
+    } = read_file(SECP256K1_SHA3_512);
+    assert!(number_of_tests >= 537, "Got unexpected number of tests");
+
+    for group in test_groups {
+        let public_key = hex::decode(group.public_key.uncompressed).unwrap();
+        assert_eq!(public_key.len(), 65);
+
+        for tc in group.tests {
+            tested += 1;
+            assert_eq!(tc.tc_id as usize, tested);
+            // eprintln!("Test case ID: {}", tc.tc_id);
+
+            match tc.result.as_str() {
+                "valid" | "acceptable" => {
+                    let message = hex::decode(tc.msg).unwrap();
+                    let message_hash = hashers::sha3_512(&message);
+                    let der_signature = hex::decode(tc.sig).unwrap();
+                    let signature = from_der(&der_signature).unwrap();
+                    let valid = secp256k1_verify(&message_hash, &signature, &public_key).unwrap();
+                    assert!(valid);
+                    if tc.comment != "k*G has a large x-coordinate" {
+                        test_secp256k1_recover_pubkey(&message_hash, &signature, &public_key);
+                    }
+                }
+                "invalid" => {
+                    let message = hex::decode(tc.msg).unwrap();
+                    let message_hash = hashers::sha3_512(&message);
+                    let der_signature = hex::decode(tc.sig).unwrap();
+
+                    if let Ok(signature) = from_der(&der_signature) {
+                        match secp256k1_verify(&message_hash, &signature, &public_key) {
+                            Ok(valid) => assert!(!valid),
+                            Err(_) => { /* this is expected for "invalid", all good */ }
+                        }
+                    } else {
+                        // invalid DER encoding, okay
+                    }
+                }
+                _ => panic!("Found unexpected result value"),
+            }
+            if tc.result == "valid" {}
+        }
+    }
+    assert_eq!(tested, number_of_tests);
+}
+
+fn test_secp256k1_recover_pubkey(message_hash: &[u8], signature: &[u8], public_key: &[u8]) {
+    // Since the recovery param is missing in the test vectors, we try both 0 and 1
+    let recovered0 = secp256k1_recover_pubkey(message_hash, signature, 0).unwrap();
+    let recovered1 = secp256k1_recover_pubkey(message_hash, signature, 1).unwrap();
+    // Got two different pubkeys. Without the recovery param, we don't know which one is the right one.
+    assert_ne!(recovered0, recovered1);
+    assert!(recovered0 == public_key || recovered1 == public_key);
+}
+
+fn from_der(data: &[u8]) -> Result<[u8; 64], String> {
```

Comment:
Good question. I think the issue is that you the test vectors make additional requirements because by nature, DER is non-deterministic, allowing signature malleability (i.e. different representations of the same signature) which allows creating multiple tx hashes from a single signature. As far as I can tell, all crypto use cases implement this themselves.

I already added the DER question here: https://github.com/CosmWasm/cosmwasm/issues/2001, so let's postpone it. For now DER only comes in through Project Wycheproof and is not something relevant for our APIs. In the Ethereum and Cosmos world DER is not used at all. Bitcoin is getting rid of it slowly.